tests/quick_tests/taskflow.cc

```diff
@@ -49,9 +49,9 @@ test1()
   B.precede(C);
 
   auto p =
-    taskflow.parallel_for(1, 11, 1, [&](int idx) { counter.fetch_add(idx); });
+    taskflow.for_each_index(1, 11, 1, [&](int idx) { counter.fetch_add(idx); });
 
-  C.precede(p.first);
+  C.precede(p);
```

Comment:
I am not sure why this fails but the dependency is a bit weird. Can you change this to 
```suggestion
  p.precede(C);
```

Reply:
The dependencies are as follows (making this clear to myself as much as to you):
* All tasks operate on the `counter` variable, which is initialized to zero.
* Tasks `A`, `B` increment `counter` by one.
* Task `C` resets the value to zero.
* `A` and `B` both precede `C`, so they count `counter` up to one, then two, and `C` resets everything to zero.
* Task `p` adds `idx` to `counter` for each `idx` between 1 and 11
* Task `C` *precedes* `p`, so what `p` does it start its sum computation from a zero state, resulting in 55 as asserted below.

In other words, the statement here looks correct in view of the assertion below. If we were to make `p` *precede* `C`, we would get some something from tasks `A`, `B`, and `p`, that is then reset to zero by task `C`. So I think that the line as stated is correct -- and also matches what you had in the previous version.

Reply:
You are right. One thing that might be problematic is the dependency to p. If you look at the generated graph
![a dia](https://user-images.githubusercontent.com/1531285/95222501-8f0f5d80-07c6-11eb-80f0-959c12110365.png)
I am not sure if the individual pfg tasks have to wait to start or not.

Maybe we should simplify this test?

Reply:
I take that back: I am pretty sure that what we are doing is legal.